packages/amplication-client/src/Resource/resourceSettings/DirectoriesProjectConfigurationSettingsForm.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  Snackbar,
+  Panel,
+  EnumPanelStyle,
+  TextField,
+} from "@amplication/design-system";
+import { gql, useMutation, useQuery } from "@apollo/client";
+import { Form, Formik } from "formik";
+import React, { useContext, useEffect } from "react";
+import * as models from "../../models";
+import { useTracking } from "../../util/analytics";
+import { formatError } from "../../util/error";
+import FormikAutoSave from "../../util/formikAutoSave";
+import { validate } from "../../util/formikValidateJsonSchema";
+import "./GenerationSettingsForm.scss";
+import { AppContext } from "../../context/appContext";
+import useProjectConfigSettingsHook from "../useProjectConfigSettingsHook";
+
+type TData = {
+  updateProjectConfigurationSettings: models.ProjectConfigurationSettings;
+};
+
+const CLASS_NAME = "generation-settings-form";
+
+const DirectoriesProjectConfigurationSettingsForm: React.FC<{}> = () => {
+  const { currentResource, addBlock } = useContext(AppContext);
+  const { data, error } = useQuery<{
+    projectConfigurationSettings: models.ProjectConfigurationSettings;
+  }>(GET_PROJECT_CONFIG_SETTINGS, {
+    variables: {
+      id: currentResource?.id,
+    },
+  });
+  const { trackEvent } = useTracking();
+
+  const [updateResourceSettings, { error: updateError }] = useMutation<TData>(
+    UPDATE_PROJECT_CONFIG_SETTINGS,
+    {
+      onCompleted: (data) => {
+        console.log({ data });
+        addBlock(data.updateProjectConfigurationSettings.id);
+      },
+    }
+  );
+  const resourceId = currentResource?.id;
+
+  useEffect(() => {
+    if (!currentResource || !resourceId) {
+      return;
+    }
+  }, [currentResource, resourceId]);
+
+  const {
+    handleSubmit,
+    PROJECT_CONFIG_FORM_SCHEMA,
+  } = useProjectConfigSettingsHook({
```

Comment:
if you added hook to this, so you can take all queries/mutation to the hook and expose it through it